deployment/clouddeploy/deployer.go

```diff
@@ -515,6 +769,31 @@ func (p *Deployer) NewCluster(ctx context.Context, def *clusterdef.Cluster) (dep
 	return thisCluster, nil
 }
 
+func (p *Deployer) NewCluster(ctx context.Context, def *clusterdef.Cluster) (deployment.ClusterInfo, error) {
+	var (
+		clusterVersion = ""
+		serverImage    = ""
+	)
+	// Ensure all node groups have the same version and image
+	for _, nodeGroup := range def.NodeGroups {
+		if clusterVersion == "" {
+			clusterVersion = nodeGroup.Version
+			serverImage = nodeGroup.Cloud.ServerImage
+		} else {
+			if clusterVersion != nodeGroup.Version || serverImage != nodeGroup.Cloud.ServerImage {
+				return nil, errors.New("all node groups must have the same version and image")
+			}
+		}
+	}
+
+	// Deploy cluster based on presence of server image
+	if serverImage != "" {
```

Comment:
I'm not necessarily opposed to this, since it's abstracted behind the Deployer interface, but is there any reason to keep the CreateNewCluster style, or can we just broadly always use deployNewCluster since the override section is optional.

Reply:
actually there is different key and values that gets passed in both api's request body. Like Package and Plan, and both have different values, like Developer Pro and developerPro and cluster provider have different values that gets passed like aws and hostedAWS, hostedAzure. And there is extra key in deploy api ie tenant id that needs to be passed. 

I think i should remove omitempty from override section and keep both request body's?

Reply:
I think that "Package" and "Plan" are items that are broadly abstracted away inside of dinocluster (you don't specify them directly), so having them be generated differently is probably possible.  Additionally, the cluster provider values are the same in that they are handled by dinocluster directly.  I think the tenantID is just an organizationID (but under a different name).  I think if the 'override' section were optional, it would be possible to broadly use that API for both serverImage and non-serverImage deployments.  It may be a worth a quick query to the cloud team to see whether it's worthwhile to support two distinct deployment mechanisms here, or if we can consolidate on just one (which would be better for simplicity and testing of dinocluster).

Reply:
I checked with cloud team, it is advisable to use deploy api but i think in cbdyn also we are using both paths, i am guessing this as results seen while creating cluster using cbdyncluster. 

Another point worth mentioning is that while we're required to provide a serverImage parameter when deploying via the /deploy API, this requirement doesn't apply to other api. Therefore, we should ensure that cbdino supports both approaches to creating clusters: with and without specifying a serverImage.

Reply:
@brett19 I ran /deploy api without passing override, and getting422 unprocessable Entity,
```
{
    "errorType": "ErrVersionsUnsupportedVersion",
    "message": "Unable to process request. The requested Couchbase version 7.1.3 is not supported in Capella. The current supported versions are 7.2,7.6. Please correct the version and try again."
}
```
And by passing server version 7.2 or 7.6, getting 500 internal server error,

`{
    "errorType": "InternalServerError",
    "message": "Oops, something went wrong. Try again later."
}`

I think it is necessary to pass override section in deploy api.

Reply:
I had asked with cloud team that if it is necessary to pass override in request body, but haven't got proper response yet, will check with them again

Reply:
Nice research.  You're point about the deploy endpoint requiring a server-image makes a good point.